client/app/components/TagsList.jsx

```diff
@@ -0,0 +1,83 @@
+import { map } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import getTags from '@/services/getTags';
+
+export class TagsList extends React.Component {
+  static propTypes = {
+    tagsUrl: PropTypes.string.isRequired,
+    onUpdate: PropTypes.func,
+  };
+
+  static defaultProps = {
+    onUpdate: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // An array of objects that with the name and count of the tagged items
+      allTags: [],
+      // A set of tag names
+      selectedTags: new Set(),
+    };
+  }
+
+  componentDidMount() {
+    getTags(this.props.tagsUrl).then((allTags) => {
+      this.setState({ allTags });
+    });
+  }
```

Comment:
An opportunity for _this_  neat syntax 😋
```js
async componentDidMount() {
  this.setState = await getTags(this.props.tagsUrl);
}
```

Reply:
Try to look at the code generated for this by `babel` - probably you'll be surprised

Reply:
YES! I wanted to suggest this in another place in this PR, but was too lazy :)